### DISCUSSION TITLE
TaskList Node

### DISCUSSION THREAD

**Discussion by JValGLZ** [eclipse-glsp/glsp#1247] [General] [answered]
Hello, I understand that the TaskList node is made out of a label, rectangle, and four circles on the GLSP java-emf-theia template. I wanted to know where it is that the tasklist-node class is being used to put the label, rectangles, and circles together so that they are rendered together. I imagine it would be in the client. Please correct me on any misconceptions. Thank you.

**Comment by planger** [comment, marked as answer]
Hi,

the structure of the diagram (nodes, child nodes, edges, etc.) is constructed in the [`TaskListGModelFactory`](https://github.com/eclipse-glsp/glsp-examples/blob/master/project-templates/java-emf-theia/glsp-server/src/main/java/org/eclipse/glsp/example/javaemf/model/TaskListGModelFactory.java). The client then just receives this GModel and maps it onto SVG for rendering.

The responsibilities of each artifact/component (source model storage, GModelFactory, view implementations of the rendering in the client) etc. are described in the [documentation](https://eclipse.dev/glsp/documentation/). Hopefully this helps to understand what the role of each component.

PS: The GLSP website seems to be down currently. There is probably some outage in the Eclipse infrastructure. I hope this is resolved soon.

**Reply by planger** [reply]
Eclipse Infra status: https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/4287
